src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -298,7 +315,7 @@ public async Task<IAsyncCursor<TResult>> ExecuteAsync(IReadBinding binding, Canc
             using (var channelBinding = new ChannelReadBinding(channelSource.Server, channel, binding.ReadPreference, binding.Session.Fork()))
             {
                 cursor = await ResumeAsync(channelBinding, cancellationToken).ConfigureAwait(false);
-                if (_startAtOperationTime == null && _resumeAfter == null)
+                if (_startAtOperationTime == null && _resumeAfter == null && _startAfter == null)
```

Comment:
Similar to above:

    cursor = await ExecuteAggregateOperationAsync(channelBinding, resuming: false, cancellationToken).ConfigureAwait(false);
    SaveInitialOperationTimeIfRequired(channel, channelBinding, cursor);

Reply:
Done